Plugin/compiler.cpp

```diff
@@ -634,6 +633,7 @@ void Compiler::AddDefaultGnuComplierOptions()
     AddCompilerOption("-pg", "Profile code when executed");
     AddCompilerOption("-w", "Inhibit all warning messages");
     AddCompilerOption("-std=c99", "Enable ANSI C99 features");
+    AddCompilerOption("-std=c11", "Enable ANSI C11 features");

```

Comment:
I think that only C89/C90 should be called "ANSI C", all later revisions just C.

Also, since you added C11 as a predefined compiler option, could you please add C17 as well?

Reply:
> I think that only C89/C90 should be called "ANSI C", all later revisions just C.

Done, they're now `Enable C99/C11 features`.

> Also, since you added C11 as a predefined compiler option, could you please add C17 as well?

C17 is a minor revision containing bugfixes only. The [GCC manual](https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html) states that those fixes are applied to C11 as well. So I'll leave it unchanged for now.

Reply:
I have read the GCC manual as well, and you are right that since the C17 revision is so small, the GCC devs backported the changes to C11. However, GCC is not the only popular compiler oficially supported by CodeLite, and e.g. in case of Clang, I haven't found out whether the Clang devs also backported the changes from C17 to C11 or not. That's why I asked you the question in the first place.

Reply:
I have read the GCC manual as well, and you are right that since the C17 revision is so small, the GCC devs backported the changes to C11. However, GCC is not the only popular compiler oficially supported by CodeLite, and e.g. in case of Clang, I haven't found out whether the Clang devs also backported the changes from C17 to C11 or not. That's why I asked you the question in the first place.

Oh, and thank you for your work, I really appreciate it :)

Reply:
> I haven't found out whether the Clang devs also backported the changes from C17 to C11 or not.

I guess Clang does the same as GCC. I've looked at the Clang repository a bit and there were no differences except for `__STDC_VERSION__`.
Also, CodeLite still allows you to set custom compliler options, so the user can easily switch to C17 anytime if really needed.